ts/Grid/Pro/CellRendering/ContentTypes/SelectContent.ts

```diff
@@ -107,8 +108,17 @@ class SelectContent extends CellContentPro implements EditModeContent {
         this.select?.remove();
     }
 
-    public getValue(): string {
-        return this.select.value;
+    public getValue(): DataTable.CellType {
+        const val = this.select.value;
+        switch (this.cell.column.dataType) {
+            case 'datetime':
+            case 'number':
+                return +val;
+            case 'boolean':
+                return val;
+            case 'string':
+                return '' + val;
+        }
```

Comment:
I would extract it to the common function. It's duplicated in textInput too. 

In select you can override the method.